app/components/action-button.js

```diff
@@ -27,6 +39,17 @@ const ActionButton = props => {
       )
     if (typeof action === 'function')
       return <Button className={cx(classes.actionButton, className)} onDone={action} {...otherProps}></Button>
+  }
+
+  if (iconName) {
```

Comment:
whether or not the icon is there, the action could still be a string, function, or empty.  The action past needs to work the same whether or not an Icon is added. 

I think the icon needs to be worked into the existing cases, rather than creating a new case just for the icon.